qiskit_experiments/analysis/curve_analysis.py

```diff
@@ -631,39 +668,56 @@ def _dictionarize(parameter_name):
 
         return fitter_options
 
-    def _subset_data(
+    @property
+    def _num_qubits(self):
+        """Getter for qubit number."""
+        return len(self.__qubits)
+
+    @property
+    def _physical_qubits(self):
+        """Getter for physical qubit indices."""
+        return list(self.__qubits)
+
+    def _data(
         self,
-        name: str,
-        data_index: np.ndarray,
-        x_values: np.ndarray,
-        y_values: np.ndarray,
-        y_sigmas: np.ndarray,
-    ) -> Tuple[np.ndarray, ...]:
-        """A helper method to extract reduced set of data.
+        series_name: Optional[str] = None,
+        label: Optional[str] = "fit_ready",
+    ) -> CurveData:
+        """Getter for experiment data set.
 
         Args:
-            name: Series name to search for.
-            data_index: An integer array representing a mapping of data location to series index.
-            x_values: Full data set of x values.
-            y_values: Full data set of y values.
-            y_sigmas: Full data set of y sigmas.
+            series_name: Series name to search for.
+            label: Label attached to data set. By default it returns "fit_ready" data.
 
         Returns:
-            Tuple of x values, y values, y sigmas for the specific series.
+            Filtered curve data set.
 
         Raises:
-            AnalysisError:
-                - When name is not defined in the __series__ definition.
+            AnalysisError: When requested series or label are not defined.
         """
-        for idx, series_def in enumerate(self.__series__):
-            if series_def.name == name:
-                sub_x_values = x_values[data_index == idx]
-                sub_y_values = y_values[data_index == idx]
-                sub_y_sigmas = y_sigmas[data_index == idx]
+        # pylint: disable = undefined-loop-variable
+        for data in self.__processed_data_set:
+            if data.label == label:
+                break
+        else:
+            raise AnalysisError(f"Requested data with label {label} does not exist.")
```

Comment:
This looks like an indentation error? Might be good to cover this with a test.
```suggestion
        for data in self.__processed_data_set:
            if data.label == label:
                break
            else:
                raise AnalysisError(f"Requested data with label {label} does not exist.")
```

Reply:
No, this is intentional. It says raising an error if the loop is not broken (for-else).
https://book.pythontips.com/en/latest/for_-_else.html#else-clause